tailwind-plugins/site-nav.js

```diff
@@ -0,0 +1,58 @@
+const plugin = require("tailwindcss/plugin");
```

Comment:
I'm creating this `site-nav.js` Tailwind plugin in this PR so we can create new utility class for the nav work. Once we can fully sunset the old nav, my plan is to clean up the template code a bit by converting the loooong class lists into a new class defined in this file.